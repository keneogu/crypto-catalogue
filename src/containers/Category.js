import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getCoin } from '../actions';
import coinAPI from '../API/api';

function Category() {
  const coins = useSelector((state) => state.coin);
  const dispatch = useDispatch();

  const fetchCoins = async () => {
    const response = await coinAPI();
    dispatch(getCoin(response.data));
  };

  return (
    <>
      <div>
        {filteredCoins.map((c) => (
          <p key={c.name}>{c.name}</p>
        ))}
      </div>
    </>
  );
}

export default Category;
