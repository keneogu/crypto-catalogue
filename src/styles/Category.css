.coin-container {
	display: flex;
	justify-content: center;
}

.coin-row {
	display: flex;
	flex-direction: row;
	justify-items: flex-start;
	align-items: center;
	height: 70px;
	border-bottom: 1px solid #eee;
	width: 900px;
}

.coin {
	display: flex;
	justify-content: space-between;
	justify-items: flex-start;
	align-items: center;
	width: 100%;
}

.coin-head {
	display: flex;
	align-items: center;
	width: 500px;
}

.coin-link {
	text-decoration: none;
	color: #474747;
}

.coin-name {
	width: auto;
	font-family: 'Roboto', sans-serif;
	font-weight: 400;
	font-size: 1.12em;
	margin-left: 10px;
}

.coin-view {
	border: 1px solid #eee;
	border-radius: 10px;
	color: rgb(185, 130, 11);
	padding: 10px;
}

.coin-head img {
	height: 30px;
	width: 30px;
}

.coin-symbol {
	text-transform: uppercase;
}

.red {
	color: #f00606;
}

.green {
	color: #11d811;
}