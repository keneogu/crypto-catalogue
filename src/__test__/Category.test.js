import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import Category from '../containers/Category';
import store from '../store';

configure({ adapter: new Adapter() });

it('matches the coin category snapshot', () => {
  const tree = renderer.create(<Provider store={store}><Category /></Provider>);
  expect(tree).toMatchSnapshot();
});

it('contains the <Filter /> component', () => {
  const component = shallow(<Provider store={store}><Category /></Provider>);
  expect(component.find('Filter')).toBeTruthy();
});
