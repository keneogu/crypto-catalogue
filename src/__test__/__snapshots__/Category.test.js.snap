// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the coin category snapshot 1`] = `
Array [
  <div
    className="coin-search"
  >
    <form
      className="search-form"
    >
      <input
        className="coin-input"
        onChange={[Function]}
        placeholder="search coin"
        type="text"
      />
    </form>
    <h3>
      search coin here...
    </h3>
  </div>,
  <div />,
]
`;
